src/Coders/Model/Relations/RelationHelper.php

```diff
@@ -24,7 +24,7 @@ public static function stripSuffixFromForeignKey($usesSnakeAttributes, $primaryK
             return preg_replace('/(_)(' . $primaryKey . '|' . $lowerPrimaryKey . ')$/', '', $foreignKey);
         } else {
             $studlyPrimaryKey = Str::studly($primaryKey);
-            return preg_replace('/(' . $primaryKey . '|' . $studlyPrimaryKey . ')$/', '', $foreignKey);
+            return preg_replace('/(_)(' . $primaryKey . '|' . $studlyPrimaryKey . ')$/', '', $foreignKey);
```

Comment:
This is a breaking change

Reply:
as i mentioned before this change fixes the User::comments()  relation example cause in the HasMany Relation class we have this condition "if (Str::snake($relationName) === Str::snake($this->parent->getClassName()))" to check the foreign key name
before my change it returns "user_ === user" because the old code does not replace the "_" from the foreign key , we need to replace _id or Id from the foreign key to generate the relation name with plural of the other table (comments not userComments)

Reply:
Thanks for clarifying, @khattab17. It does [break some of our tests](https://travis-ci.org/github/reliese/laravel/jobs/761748708) though. Which means it is introducing some unexpected behaviour as well. If you run `phpunit`, you might be able to see what's going on.

Reply:
i fixed this bug , i think now it passes the phpunit test